pkg/schedule/affinity/labeler.go

```diff
@@ -293,21 +300,31 @@ func (m *Manager) updateAffinityGroupPeersWithAffinityVer(groupID string, affini
 		return nil, err
 	}
 
-	// Step 1: Check whether the Group exists and validate affinityVer.
+	// Step 1: Check whether the Group exists and validate.
 	m.metaMutex.Lock()
 	defer m.metaMutex.Unlock()
 	group := m.GetAffinityGroupState(groupID)
-	if group == nil || (affinityVer != 0 && group.affinityVer != affinityVer) {
-		if affinityVer != 0 {
-			// No error is generated for changes with a non-zero affinityVer.
-			return nil, nil
-		}
+	if group == nil {
 		return nil, errs.ErrAffinityGroupNotFound.GenWithStackByArgs(groupID)
 	}
+	// When affinityVer is non-zero, it indicates a non-admin operation and triggers additional checks.
+	// Note: if the check fails, no changes are applied. The existing Group is returned as-is without error.
+	if affinityVer != 0 {
+		// If affinityVer is not equal, the update may come from stale statistics and will be ignored.
+		if group.affinityVer != affinityVer {
+			return group, nil
+		}
+		// Group must not change voterStoreIDs while it is not in the expired state.
+		// RegularSchedulingEnabled == IsExpired
+		// The VoterStoreIDs from the API and RegionInfo are already sorted, so they can be compared directly
```

Comment:
I think it is better to sort again in this function. I am not sure it is sorted in `observe`
```
voterStoreIDs := make([]uint64, 0, len(region.GetVoters()))
	for _, voter := range region.GetVoters() {
		voterStoreIDs = append(voterStoreIDs, voter.GetStoreId())
	}
```